packages/core-sdk/lib/Connectors/BaseConnector.ts

```diff
@@ -1,27 +1,64 @@
 import { ConnectorEvent, IConnectorAdapters } from "../Types";
 import EventEmitter from "eventemitter3";
-import { IConnector } from "../Types/IConnector";
+import { IConnector, OnNetworkChangeOptions } from "../Types/IConnector";
 import { Callback } from "../Utils/Typings";
 import { IBlockchainConfig } from "../Types/IBlockchainConfig";
 import { IConnectorMetadata } from "../Types";
+import { blockchainConfigMap } from "../Blockchains";
 
 export abstract class BaseConnector implements IConnector {
   protected emitter = new EventEmitter<ConnectorEvent>();
   _adapter: IConnectorAdapters | undefined;
 
   constructor(
     public readonly metadata: IConnectorMetadata,
-    public readonly config: IBlockchainConfig
+    public config: IBlockchainConfig
   ) {}
 
-  protected abstract _connect(): Promise<IConnectorAdapters>;
+  protected abstract _connect(
+    config?: IBlockchainConfig
```

Comment:
connect() is just a wrapper of _connect() to do extra logic so they should need to have the same options property. 
Why do you need to pass the config to the _connect method?